File: apps/cargo-tracker/src/main/java/com/example/cargotracker/shared/domain/model/ShipperId.java
package com.example.cargotracker.shared.domain.model;

import java.util.Objects;
import java.util.UUID;

/**
 * 荷主識別子。
 * Shared Kernel として booking / shipper の両コンテキストから参照される。
 */
public final class ShipperId {

    private final UUID value;

    public ShipperId(UUID value) {
        if (value == null) {
            throw new IllegalArgumentException("ShipperId の値は null にできません");
        }
        this.value = value;
    }

    public static ShipperId generate() {
        return new ShipperId(UUID.randomUUID());
    }

    public UUID value() {
        return value;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof ShipperId that)) return false;
        return Objects.equals(value, that.value);
    }

    @Override
    public int hashCode() {
        return Objects.hashCode(value);
    }

    @Override
    public String toString() {
        return value.toString();
    }
}

File: apps/cargo-tracker/internal/shared/domain/shipper_id.go
// Package domain は全コンテキストで共有する共有カーネル（Shared Kernel）を提供する。
package domain

import "errors"

// ErrEmptyShipperId は ShipperId が空の場合に返される。
var ErrEmptyShipperId = errors.New("shipper id must not be empty")

// ShipperId は荷主を一意に識別する共有カーネルの値オブジェクト。
// Booking Context と Shipper Context で共有する。
type ShipperId struct {
	value string
}

// NewShipperId はバリデーション付きで ShipperId を生成する。
func NewShipperId(value string) (ShipperId, error) {
	if value == "" {
		return ShipperId{}, ErrEmptyShipperId
	}
	return ShipperId{value: value}, nil
}

// Value は ShipperId の文字列表現を返す。
func (s ShipperId) Value() string { return s.value }
